src/Microsoft.Azure.Batch.SoftwareEntitlement.Server/Startup.cs

```diff
@@ -31,6 +32,17 @@ public void ConfigureServices(IServiceCollection services)
         {
             services.AddMvc();
             services.TryAddSingleton<IHttpContextAccessor, HttpContextAccessor>();
+            services.TryAddSingleton(provider =>
+            {
+                var serverOptions = provider.GetService<SoftwareEntitlementsController.ServerOptions>();
+                return new NodeEntitlementReader(
+                    serverOptions.Audience,
+                    serverOptions.Issuer,
+                    serverOptions.SigningKey,
+                    serverOptions.EncryptionKey);
+            });
```

Comment:
It would be better if the `NodeEntitlementReader` accepted the `ServerOptions` instance directly, and if we left the order of object resolution to the container.

Reply:
The reason that `NodeEntitlementReader` doesn't accept a `ServerOptions` is that `ServerOptions` is defined in the `Controller` class. The main argument against moving it is probably the `ServerFlags` property, which is specific to the `Controller` and doesn't really belong in the `SoftwareEntitlement` project.

Reply:
🤔